spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/SpannerOperations.java

```diff
@@ -0,0 +1,144 @@
+	 */
+	DatabaseClient getDatabaseClient();
+
+	/**
+	 * Finds a single stored object using a key.
+	 * @param entityClass the type of the object to retrieve.
+	 * @param key the key of the object.
+	 * @param <T> the type of the object to retrieve.
+	 * @return an object of the requested type. Returns null if no object could be found
+	 * stored with the given key.
+	 */
+	<T> T find(Class<T> entityClass, Key key);
+
+	/**
+	 * Finds objects stored from their keys.
+	 * @param entityClass the type of the object to retrieve.
+	 * @param keys the keys of the objects to retrieve.
+	 * @param options Spanner read options with which to conduct the read operation.
+	 * @param <T> the type of the object to retrieve.
+	 * @return a list of objects that could be found using the given keys. If no keys
+	 * could be found the list will be empty.
+	 */
+	<T> List<T> find(Class<T> entityClass, KeySet keys, Options.ReadOption... options);
+
+	/**
+	 * Finds objects by using an SQL statement.
+	 * @param entityClass the type of object to retrieve.
+	 * @param statement the SQL statement used to select the objects.
+	 * @param options Spanner read options with which to conduct the read operation.
+	 * @param <T> the type of object to retrieve.
+	 * @return a list of the objects found. If no keys could be found the list will be
+	 * empty.
+	 */
+	<T> List<T> find(Class<T> entityClass, Statement statement,
+			Options.QueryOption... options);
+
+	/**
+	 * Finds all objects of the given type.
+	 * @param entityClass the type of the object to retrieve.
+	 * @param options Spanner read options with which to conduct the read operation.
+	 * @param <T> the type of the object to retrieve.
+	 * @return a list of all objects stored of the given type. If there are no objects an
+	 * empty list is returned.
+	 */
+	<T> List<T> findAll(Class<T> entityClass, Options.ReadOption... options);
+
+	/**
+	 * Deletes an object based on a key.
+	 * @param entityClass the type of the object to delete.
+	 * @param key the key of the object to delete from storage.
+	 */
+	void delete(Class entityClass, Key key);
+
+	/**
+	 * Deletes an object from storage.
+	 * @param object the object to delete from storage.
+	 */
+	void delete(Object object);
+
+	/**
+	 * Deletes multiple objects from storage.
+	 * @param entityClass the type of the object to delete.
+	 * @param objects the objects to delete.
+	 * @param <T> the type of the object to delete.
+	 */
+	<T> void delete(Class<T> entityClass, Iterable<? extends T> objects);
+
+	/**
+	 * Deletes objects given a set of keys.
+	 * @param entityClass the type of object to delete.
+	 * @param keys the keys of the objects to delete.
+	 */
+	void delete(Class entityClass, KeySet keys);
+
+	/**
+	 * Insert an object into storage.
+	 * @param object the object to insert.
+	 */
+	void insert(Object object);
+
+	/**
+	 * Update an object already in storage.
+	 * @param object the object to update.
+	 */
+	void update(Object object);
```

Comment:
Will add.

Reply:
Actually, there is an update from what I see.

Reply:
`SpannerOperations.update(T object, String... properties)`? I don't see it.

Reply:
I don't follow...

there's this in SpannerMutationFactory
```	
<T> Mutation update(T object, String... properties);
```

then there's this in SpannerOptions
```
void update(Object object);
```

then there's the usage of the factory method in SpannerTemplate
```
	@Override
	public void update(Object object) {
		Mutation mutation = this.mutationFactory.update(object);
		this.databaseClient.write(Arrays.asList(mutation));
	}
```

Reply:
Where are you passing the `properties` to update in this example? The user might want to update only certain properties of the object that they are passing to `SpannerTemplate.update()`.

Reply:
Ooh yes, I had missed the Properties part. Will add.

Reply:
I decided to remove being able to specify this in the update command actually. I presume the purpose of letting people specify the specific columns they want to update is so they can set those values without having to populate the values they don't want to worry about. However I think they should just be doing that kind of thing outside of framework, like via an actual SQL query. it would lead to objects getting out of sync with what's stored in Spanner, too.